de.peeeq.wurstscript/src/main/java/de/peeeq/wurstio/languageserver/requests/MapRequest.java

```diff
@@ -344,37 +348,43 @@ protected File compileScript(WurstGui gui, ModelManager modelManager, List<Strin
         return compileMap(modelManager.getProjectPath(), gui, mapCopy, runArgs, model);
     }
 
-    protected File compileScript(ModelManager modelManager, WurstGui gui, @Nullable File testMap) throws Exception {
-        if (testMap != null && testMap.exists()) {
-            boolean deleteOk = testMap.delete();
+    protected File compileScript(ModelManager modelManager, WurstGui gui, Optional<File> testMap) throws Exception {
+        if (testMap.isPresent() && testMap.get().exists()) {
+            boolean deleteOk = testMap.get().delete();
             if (!deleteOk) {
                 throw new RequestFailedException(MessageType.Error, "Could not delete old mapfile: " + testMap);
             }
         }
-        if (map != null) {
-            Files.copy(map, testMap);
+        if (map.isPresent() && testMap.isPresent()) {
+            Files.copy(map.get(), testMap.get());
         }
 
-        parseCustomPatchVersion();
-
         // first compile the script:
         File compiledScript = compileScript(gui, modelManager, compileArgs, testMap);
 
-        WurstModel model = modelManager.getModel();
-        if (model == null || model.stream().noneMatch((CompilationUnit cu) -> cu.getCuInfo().getFile().endsWith("war3map.j"))) {
+        Optional<WurstModel> model = Optional.ofNullable(modelManager.getModel());
+        if (!model.isPresent()
+                || model
+                    .get()
+                    .stream()
+                    .noneMatch((CompilationUnit cu) -> cu.getCuInfo().getFile().endsWith("war3map.j"))) {
             println("No 'war3map.j' file could be found inside the map nor inside the wurst folder");
             println("If you compile the map with WurstPack once, this file should be in your wurst-folder. ");
             println("We will try to start the map now, but it will probably fail. ");
         }
         return compiledScript;
     }
 
-    private void parseCustomPatchVersion() {
-        if (wc3Path != null) {
-            W3Utils.parsePatchVersion(new File(wc3Path));
-            if (W3Utils.getWc3PatchVersion() == null) {
+    private W3Utils parseCustomPatchVersion() throws RequestFailedException {
```

Comment:
if `W3Utils` is not gonna be singleton, then rename it to something proper (e.g. W3InstallationData or so) and probably also rename the method

Reply:
Thanks! Fixed